tests/integ/modin/groupby/test_groupby_size.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+import modin.pandas as pd
+import numpy as np
+import pytest
+
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from tests.integ.modin.sql_counter import SqlCounter
+from tests.integ.modin.utils import eval_snowpark_pandas_result
+
+
+@pytest.mark.parametrize(
+    "by",
+    [
+        "col1_grp",
+        "col2_int64",
+        "col3_int_identical",
+        "col4_int32",
+        "col5_int16",
+        "col6_mixed",
+        "col7_bool",
+        "col8_bool_missing",
+        "col9_int_missing",
+        "col10_mixed_missing",
+        ["col1_grp", "col2_int64"],
+        ["col6_mixed", "col7_bool", "col3_int_identical"],
+    ],
+)
+@pytest.mark.parametrize("as_index", [True, False])
+def test_groupby_size(by, as_index):
+    snowpark_pandas_df = pd.DataFrame(
+        {
+            "col1_grp": ["g1", "g2", "g0", "g0", "g2", "g3", "g0", "g2", "g3"],
+            "col2_int64": np.arange(9, dtype="int64") // 3,
+            "col3_int_identical": [2] * 9,
+            "col4_int32": np.arange(9, dtype="int32") // 4,
+            "col5_int16": np.arange(9, dtype="int16") // 3,
+            "col6_mixed": np.concatenate(
+                [
+                    np.arange(3, dtype="int64") // 3,
+                    np.arange(3, dtype="int32") // 3,
+                    np.arange(3, dtype="int16") // 3,
+                ]
+            ),
+            "col7_bool": [True] * 5 + [False] * 4,
+            "col8_bool_missing": [
+                True,
+                None,
+                False,
+                False,
+                None,
+                None,
+                True,
+                False,
+                None,
+            ],
+            "col9_int_missing": [5, 6, np.nan, 2, 1, np.nan, 5, np.nan, np.nan],
+            "col10_mixed_missing": np.concatenate(
+                [
+                    np.arange(2, dtype="int64") // 3,
+                    [np.nan],
+                    np.arange(2, dtype="int32") // 3,
+                    [np.nan],
+                    np.arange(2, dtype="int16") // 3,
+                    [np.nan],
+                ]
+            ),
+        }
+    )
+    pandas_df = snowpark_pandas_df.to_pandas()
+    with SqlCounter(query_count=1 if as_index else 2):
```

Comment:
What causes the extra query when `as_index=False`?

Reply:
Looks like there's an extra `count(1)` query being issued.

Reply:
I think we have this same behavior (extra count query) for reset_index, so I believe it is to be expected

Reply:
It's coming from the `squeeze(axis=1)` call here: https://github.com/snowflakedb/snowpark-python/pull/1763/files#diff-1d632233293b5d0c0f50ad639e2415f3d32851fd41d59f2295c38e29ff0e3707R753, but it shouldn't, `axis=1` does not need the `len` of the dataframe). We should fix that bug in another PR.

Reply:
I think this is `columnarize` in the `Series` constructor again. We need to kill it: https://github.com/modin-project/modin/issues/6111

Reply:
https://github.com/snowflakedb/snowpark-python/pull/1767

Reply:
That fixes the issue, it's not columnarize in this case 😄